### ISSUE TITLE
Bump & Tilt: Use accelerometer to model physically nudging a pinball machine

### ISSUE THREAD

**Issue by hackerb9** [dozingcat/Vector-Pinball#100]
I instinctively jostle my phone to the side when the ball is balanced precariously on an edge. 

This ought to affect the ball, but currently does not.

Of course, shaking the phone too much should cause a tilt. Perhaps tilt detection would also be randomized somewhat, weighted by the magnitude of acceleration.

The main challenge with this is that I'm not sure of the physics equation since one is technically moving the table. I could be wrong, but I believe that if the ball is next to a wall, it would act a bit like a flipper. If not, the mass of the ball would keep it in place as the table moves underneath it and back. That could be modeled relative to the table by simply putting a force on the ball opposite of the direction the phone was bumped (and then back). But I think it would look cool if one could actually see the table shake; that is, move the playing field, not the ball.

Again, I might be wrong about how the physics work. I think a simple impulse is sufficient, but maybe the shaking is best modeled as an oscillating spring with a damper? 

However, whatever equations are used, I think adding some physicality would be a great addition to Vector Pinball.

**Comment by maverick74** [comment]
I vote for this as well!

Tilt is something iconic!

**Comment by Jacques-Olivier-Farcy** [comment]
Yes it is a great idea, as i have made a vector wood pinball it would be really fun.

![image](https://user-images.githubusercontent.com/19731340/199671457-75658a9b-6f9a-4e47-b859-d1b71513bbe5.png)

**Comment by hackerb9** [comment]
I looked into it a bit and the proper name for this is **nudging**. It sounds like the consensus among experts is, nudging is not just a part of pinball, nudging _is_ pinball. It's even more important to playing then being able to accurately aim.

Also, I saw a YouTube video where a champion was attempting to explain nudging. Mostly they failed, saying you just have to play to understand, but they did at least mention that the guess I had before was right: sometimes you nudge in order to push a bumper on the table into the ball,  like when the ball is about to go down one of the side drains. Other times, you nudge to change the course of the ball, like when the ball is falling down the center and can't be reached by your flippers. 

I believe the same equation for moving the table under the ball will produce both effects.

**Comment by hackerb9** [comment]
Some considerations: 

* Should Vector Pinball's table physics directly map the table to what is actually happening with the device? That would be simplest, and may feel most readily natural. 
* Alternately, should only the bottom of the table be affected by nudging, as it would be on a real machine. (Imagine the Android device is just the part of the machine that one holds when playing: the bottom corners, with the flippers.
* Will the nudging effect need be damped down for small motions caused just by a person holding a phone or tilting it while playing? Hopefully not, as that may make it feel unrealistic.
* If the accelerometers are mapped to the entire table and not just the bottom, should table motions which are not possible in a real pinball machine be somehow excluded? Again, hopefully not, but it may not feel "right" to more expert players if they can nudge the top of the table.

**Comment by maverick74** [comment]
I would vote for everything "the most real possible"!

About nudging all device vs just the bottom i would leave options for both. If only one option is possible, then i would go with "all device" :)

**Comment by Jacques-Olivier-Farcy** [comment]
And nudging should be working also on "physical" pinball with a specific sensor.
